docs/client-sdk/javascript/tutorials/build-an-xmtp-hello-world-app.md

```diff
@@ -0,0 +1,576 @@
+     walletConnectProvider({ projectId: "<YOUR_PROJECT_ID>" }),
+   ]);
+   const wagmiClient = createClient({
+     autoConnect: true,
+     connectors: modalConnectors({ appName: "web3Modal", chains }),
+     provider,
+   });
+   // Web3Modal Ethereum Client
+   const ethereumClient = new EthereumClient(wagmiClient, chains);
+   // highlight-start
+   type MessageListProps = {
+     msg: DecodedMessage[];
+   };
+   // highlight-end
+
+   function App() {
+     const { isConnected } = useAccount();
+     // highlight-start
+     const [messages, setMessages] = useState<DecodedMessage[]>([]);
+     const [client, setClient] = useState<any>();
+     const [xmtpClientAddress, setXmtpClientAddress] = useState<any>();
+     // highlight-end
+
+     // highlight-start
+     const initXmtp = async function () {
+       const signer = await fetchSigner();
+       const xmtp = await Client.create(signer, { env: "production" });
+       const conversation = await xmtp.conversations.newConversation(PEER_ADDRESS);
+       const messages = await conversation.messages({
+         direction: SortDirection.SORT_DIRECTION_DESCENDING,
+       });
+
+       setClient(conversation);
+       setMessages(messages);
+       setXmtpClientAddress(xmtp.address);
+       // highlight-end
+     };
+     // highlight-start
+     useEffect(() => {
+       if (xmtpClientAddress) {
+         const streamMessages = async () => {
+           const newStream = await client.streamMessages();
+           for await (const msg of newStream) {
+             setMessages((prevMessages) => {
+               const messages = [...prevMessages];
+               messages.unshift(msg);
+               return messages;
+             });
+           }
+         };
+         streamMessages();
+       }
+     }, [client, xmtpClientAddress]);
+     // highlight-end
+
+     // highlight-start
+     const onSendMessage = async () => {
+       const message = "gm XMTP bot!";
+       await client.send(message);
+     };
+    // highlight-end
+
+     // highlight-start
+     const MessageList: React.FC<MessageListProps> = ({ msg }) => {
+       return (
+         <ul>
+           {msg.map((message, index) => (
+             <li key={index}>{message.content}</li>
+           ))}
+         </ul>
+       );
+     };
+     // highlight-end
+
+     return (
+       <div className="App">
+         <Web3Button />
+         <Web3Modal
+           projectId="<YOUR_PROJECT_ID>"
+           ethereumClient={ethereumClient}
+         />
+         // highlight-start
+         {isConnected && xmtpClientAddress && (
+           <>
+             <MessageList msg={messages} />
+             <button onClick={onSendMessage}>Send GM</button>
+           </>
+         )}
+         {isConnected && !xmtpClientAddress && (
+           <button onClick={initXmtp}>Connect to XMTP</button>
+         )}
+         // highlight-end
+       </div>
+     );
+   }
+
+   export default App;
+
```

Comment:
When I do run npm start, I am seeing 126 warnings - along these lines: 

`WARNING in ./node_modules/safe-json-utils/dist/cjs/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/jha/src/my-app/node_modules/safe-json-utils/src/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/jha/src/my-app/node_modules/safe-json-utils/src/index.ts'`

If we don't think these warnings are an issue - I wonder if it might be a good idea to include a note about it? Less experienced folx might worry that something is wrong, even if they are just warnings.

Reply:
Hmm i dont see these errors on my end after these steps.

Reply:
Maybe it's just something about my local environment. They're just warnings and don't seem to be impacting the app itself - so I think it's OK! Thank you for checking on your end!